### PR TITLE
Fix movement effect warning spam due to Spearhead (XEL0306)

### PR DESCRIPTION
**Pull request by lL1l1**
## Description of the proposed changes

- Add an exception for the warning condition in case there is a Treads table in the movement effects layer group, similar to #6480.
- Fix Spearhead's treads since I noticed it

## Testing done on the proposed changes

Look at the spearhead's treads and make sure they work and aren't disconnected visually.
Checked all blueprints that could trigger the warning using this command:
```lua
UI_Lua 
for i,v in __blueprints do
	if type(i) == 'number' then continue end
	local me = v.Display.MovementEffects
	local t = {"Land","Seabed","Sub","Water","Air","Orbital"}
	for _, l in t do
		local mel = me[l]
		if mel and not mel.Effects then
			LOG(string.format('%-7s %-12s %-20s %-20s', v.FactionCategory, v.TechCategory, LOC(v.Description), LOC(v.General.UnitName) or ''))
			for i, _ in mel do
				LOG(i)
			end
		end
	end
end
```
<details> <summary> Output only has `Contrails`, `Footfall`, and `Treads` </summary>

```
AEON    EXPERIMENTAL Experimental Assault Bot Galactic Colossus   
Footfall
UEF     TECH3        Mobile Missile Platform Spearhead           
Treads
CYBRAN  TECH2        Fighter/Bomber       Corsair             
Contrails
UEF     TECH1        Interceptor          Cyclone             
Contrails
CYBRAN  EXPERIMENTAL Experimental Spiderbot Monkeylord          
Footfall
SERAPHIM TECH2        Gunship              Vulthoo             
Contrails
CYBRAN  TECH2        Destroyer            Salem Class         
Footfall
AEON    TECH2        Guided Missile       Mercy               
Contrails
CYBRAN  TECH1        Air Scout            Flying Eyes         
Contrails
UEF     TECH1        Air Scout            Hummingbird         
Contrails
CYBRAN  TECH1        Interceptor          Prowler             
Contrails
CYBRAN  EXPERIMENTAL Experimental Megabot Megalith            
Footfall
SERAPHIM TECH3        Spy Plane            Iaselen             
Contrails
```

</details>

## Checklist
- [x] Changes are annotated, including comments where useful
- [x] Changes are documented in the changelog for the next game version

### CODE REVIEW

**Comment by lL1l1** [comment]
@BlackYps Would this be possible to merge in before release? I've added its PR number to the 6436 snippet in that case, but otherwise it should be separate.

File: lua/sim/Unit.lua [hunk 1]
```diff
@@ -3729,8 +3729,8 @@ Unit = ClassUnit(moho.unit_methods, IntelComponent, VeterancyComponent, DebugUni
             local effectTypeGroups = bpTable.Effects
 
             if not effectTypeGroups or (effectTypeGroups and (table.empty(effectTypeGroups))) then
-                -- warning isn't needed if this layer's table is used for Footfall or Contrails without terrain effects
-                if not bpTable.Footfall and not bpTable.Contrails then
+                -- warning isn't needed if this layer's table is used for Footfall or Contrails or Treads without terrain movement effects
+                if not bpTable.Footfall and not bpTable.Contrails and not bpTable.Treads then
```
Comment (Garanas) [inline lua/sim/Unit.lua:3733]: At this point perhaps it is more readable as `not (X or Y or Z)`

Comment (clyfordv) [reply lua/sim/Unit.lua:3733]: I concur with Jip

**Review by clyfordv — APPROVED** [review]
Looks good, errors are eliminated (I only tested the Spearhead, not sure what other units were throwing the error, but solution applies universally).

**Pull request merged**